src/main/java/org/mskcc/cbio/oncokb/service/SlackService.java

```diff
@@ -120,6 +128,44 @@ public void sendConfirmationOnUserAcceptsTrialAgreement(UserDTO userDTO, Instant
         }
     }
 
+    public List<UserIdMessagePair> getAllUnapprovedUserRequestsSentAfter(int daysAgo) {
+        List<UserIdMessagePair> userList = new ArrayList<>();
+        Slack slack = Slack.getInstance();
+        final String REQUEST_MESSAGE_TEXT = "This content can't be displayed.";
+
+        try {
+            String daysAgoTs = Long.toString(Instant.now().getEpochSecond() - DAY_IN_SECONDS * daysAgo);
+            ConversationsHistoryResponse conversationsHistory = slack.methods().conversationsHistory(r -> r
+                .token(applicationProperties.getSlack().getSlackBotOauthToken())
+                .channel(applicationProperties.getSlack().getUserRegistrationChannelId())
+                .oldest(daysAgoTs)
+                .limit(1000)
+                .inclusive(true));
+            Collections.reverse(conversationsHistory.getMessages());
+            for (Message message : conversationsHistory.getMessages()) {
+                if (Objects.nonNull(message.getText()) && message.getText().equals(REQUEST_MESSAGE_TEXT) && Objects.nonNull(message.getBlocks())) {
+                    if (!(getBlockWithId(message.getBlocks(), COLLAPSED).isPresent()
+                        || getBlockWithId(message.getBlocks(), ACADEMIC_CLARIFICATION_NOTE).isPresent()
+                        || getBlockWithId(message.getBlocks(), MSK_USER_NOTE).isPresent()
+                        || getBlockWithId(message.getBlocks(), TRIAL_ACCOUNT_NOTE).isPresent()
+                        || getBlockWithId(message.getBlocks(), APPROVED_NOTE).isPresent()
+                        || getBlockWithId(message.getBlocks(), CONVERT_TO_REGULAR_ACCOUNT_NOTE).isPresent())
+                    ) {
+                        if (getBlockWithId(message.getBlocks(), USER_ID).isPresent()) {
+                            ContextBlock userIdBlock = (ContextBlock) getBlockWithId(message.getBlocks(), USER_ID).get();
+                            PlainTextObject userIdText = (PlainTextObject) userIdBlock.getElements().get(0);
+                            Long userId = Long.parseLong(userIdText.getText().substring(9));
+                            userList.add(new UserIdMessagePair(userId, message));
+                        }
+                    }
+                }
+            }
+        } catch (IOException | SlackApiException e) {
+            log.error("error: {}", e.getMessage(), e);
+        }
+        return userList;
+    }
+
```

Comment:
This is just rearrangement, nothing changed?

Reply:
Yes, double checked just now.